docs/guide/snaps-concepts.md

```diff
@@ -2,6 +2,128 @@
 
 [[toc]]
 
+## Accounts and Key Management
+
+The Snaps API contains functionality that allows you to manage users' private keys, with their approval. This capability comes with great responsiblity on the part of the developer: misplaced or stolen private keys will lead to a complete loss of funds for the users of the snap. Our team has developed some guidelines to help you assess the feasibility and sanity of your snap.
+
+### Key Management Guidelines
+
+The general guideline for responsible key management is:
+
+> Don't create a situation where your user can lose money
+
+In practice, here are some examples that would break the above policy:
+
+1. Allowing extraction of private keys outside the snap in any way, especially through RPC or network connections
+2. Arbitrary code execution with access to private keys.
+3. Not informing the user properly or not getting informed consent before doing a destructive operation, for example a transaction.
+4. Asking for consent but ignoring the decision.
+5. A bug that leads to any of the above.
+
+And here are some examples of abiding by the above policy:
+
+1. Manipulation of the private keys, their storage in snaps' persistent storage, without them ever leaving the Snaps Secure EcmaScript sandbox.
```

Comment:
It's not clear to me as a developer at this point what this entails. Is Snaps doing this for me? Can we link to a code sample?